src/EGraphs/extract.jl

```diff
@@ -73,27 +73,22 @@ function find_costs!(extractor::Extractor{CF,CT}) where {CF,CT}
 end
 
 """
-A basic cost function, where the computed cost is the size
-(number of children) of the current expression.
+A basic cost function, where the computed cost is the number 
+of expression tree nodes.
 """
 function astsize(n::VecExpr, op, costs::Vector{Float64})::Float64
   v_istree(n) || return 1
-  cost = 2 + v_arity(n)
-  cost + sum(costs)
+  cost = 1 + sum(costs)
 end
 
 """
-A basic cost function, where the computed cost is the size
-(number of children) of the current expression, times -1.
+A basic cost function, where the computed cost is the number
+of expression tree nodes times -1.
 Strives to get the largest expression
 """
-function astsize_inv(n::VecExpr, op, costs::Vector{Float64})::Float64
-  v_istree(n) || return -1
-  cost = -(1 + v_arity(n)) # minus sign here is the only difference vs astsize
-  cost + sum(costs)
-end
+astsize_inv(n::VecExpr, op, costs::Vector{Float64})::Float64 = -astsize(n, op, costs)
 
+function extract!(g::EGraph, costfun, cost_type = Float64)
+  Extractor(g, costfun, cost_type)()
+end
```

Comment:
I would restore the optional `cost_type` argument

Reply:
By restore, you mean remove from line 91?

I made this change because I implemented a cost function that returns a custom type (for which I implemented `isless` and `typemax`). If the two methods are implemented the extractor works with custom cost types.

`cost_type` could probably also be infered from `costfun` instead.

Reply:
Whoops sorry, I read the diff the other way around. My bad.